spec/ruby/optional/capi/numeric_spec.rb

```diff
@@ -429,4 +429,20 @@
       lambda { @s.rb_num_coerce_relop(2, obj, :<) }.should raise_error(ArgumentError)
     end
   end
+
+  describe "rb_absint_singlebit_p" do
+    it "returns 1 if absolute value fits into a bit" do
+      examples = {
+                     1 => 1,  2 => 1,  3 => 0,
+                    -1 => 1, -2 => 1, -3 => 0,
+                    bignum_value => 1,
+                    bignum_value(1) => 0,
+                    -bignum_value => 1,
+                    -bignum_value(1) => 0,
+                  }
+      examples.each do |val, res|
+        @s.rb_absint_singlebit_p(val).should == res
+      end
```

Comment:
It seems `be_computed_by` would be better here, so that the error message can include the problematic value.
Personally, I would prefer individual lines with `should` though, it's a bit repetitive but much simpler and clearer.

Reply:
I've updated to use `should`